pkg/filter/headertojson/headertojson.go

```diff
@@ -204,5 +223,6 @@ func (h *HeaderToJSON) handle(ctx context.HTTPContext) string {
 		return resultJSONEncodeDecodeErr
 	}
 	ctx.Request().SetBody(bytes.NewReader(bodyBytes))
+	ctx.Request().Header().Del("Content-Encoding")
```

Comment:
this should be fine.
but please note keep gzip enabled saves bandwidth, so another option is the compress the body again if gzip is on.

Reply:
i think maybe we can make another filter to encoding body, because if we encode body in headertojson filter, then we need decode it again for filters like Kafka...

Reply:
In my opinion, this should be a built-in feature for HTTPServer. Decompression should be handled automatically. Compression should require parameters to be specified.

Reply:
how about make compress and decompress as a new filter. if we decompress it automatically and then compress it again, it is a little bit waste...

Reply:
yes, adding a new filter is better, or we can enhance the existing `RequestAdaptor` to support compress/decompress.
the HTTPServer is to receive requests, it should not care about the content of the request, and most pipelines only process request headers, in which case decompressing the body of all requests is useless.

Reply:
sounds great, i will check `RequestAdaptor` and update it now!

Reply:
updated!